src/todo-batch-generator.ts

```diff
@@ -54,22 +45,22 @@ export default class TodoBatchGenerator {
    * Exact matches match on engine, ruleID, line and column
    * Fuzzy matches match on engine, ruleID and source
    *
-   * @param lintResults - The raw linting data.
+   * @param maybeTodos - The linting data, converted to TodoDataV2 format.
    * @param existingTodos - Existing todo lint data.
    * @returns
    */
   generate(
-    lintResults: LintResult[],
+    maybeTodos: Set<TodoDataV2>,
     existingTodos: Map<TodoFilePathHash, TodoMatcher>
   ): TodoBatches {
     const add = new Map<TodoFileHash, TodoDataV2>();
     const expired = new Map<TodoFileHash, TodoDataV2>();
     const stable = new Map<TodoFileHash, TodoDataV2>();
     let remove = new Map<TodoFileHash, TodoDataV2>();
 
-    const unmatched = buildTodoData(this.baseDir, lintResults, this.options?.todoConfig);
+    maybeTodos = new Set(maybeTodos);
```

Comment:
I'm cloning here so as to not mutate the data passed in, which allows us to verify and compare during tests.